portfolio/src/main/java/com/google/sps/servlets/AuthenticationServlet.java

```diff
@@ -0,0 +1,75 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that authenticates users using Google's Users API */
+@WebServlet("/authenticate")
+public class AuthenticationServlet extends HttpServlet {
+
+  /**
+   * On GET request, returns true if the user is currently logged in, false otherwise. Also returns
+   * a message to the user depending on whether they are logged in or not.
+   *
+   * @param request The request made by the connecting client.
+   * @param response The response that is sent back to the client.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    boolean isLoggedIn = userService.isUserLoggedIn();
+    String message;
+
+    // Create message based on whether the user is logged in or not
+    if (isLoggedIn) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      String urlToRedirectToAfterUserLogsOut = "/comments.html";
+      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+
+      message =
+        "<p>You're logged in as "
+          + userEmail
+          + ". "
+          + "Logout <a href=\\\""
+          + logoutUrl
+          + "\\\">here</a>.</p>";
+    } else {
+      String urlToRedirectToAfterUserLogsIn = "/comments.html";
+      String loginUrl = userService.createLoginURL(urlToRedirectToAfterUserLogsIn);
+
+      message =
+        "<p>Looks like you're not logged in. "
+          + "To submit a comment, login <a href=\\\""
+          + loginUrl
+          + "\\\">here</a>.</p>";
+    }
+
+    // Build json to send
```

Comment:
You can use [JSONObject](https://developer.android.com/reference/org/json/JSONObject) to make this easier. In pom.xml import with 
``` 
<dependency>
      <groupId>org.json</groupId>
      <artifactId>json</artifactId>
      <version>20090211</version>
</dependency>
```
And  `import org.json.JSONObject;`  would be the java import statement

Reply:
Thanks! I actually used a separate JSON library in a previous branch that I will merge with this one, but I'll definitely check out what different and pros/cons would exist between the two libraries.